lib/handler/RedirectHandler.js

```diff
@@ -91,6 +91,11 @@ class RedirectHandler {
       ? null
       : parseLocation(statusCode, headers)
 
+    if (this.history.length >= this.maxRedirections) {
```

Comment:
Remember to add the control flag, otherwise, this will throw every time we reach max redirections, and might not be expected by the implementer.

Reply:
thank you very much, as you said, I reviewed and improved the test and then I updated my code and added a flag 

first I solved my errors in the test/redirect-request.js file, but then I encountered an error in the test/redirect-stream.js file, I added a few logs and the error output was like this 

```js
 FAIL ​ test/redirect-stream.js
 ✖ max redirects


        this.abort(new Error('max redirects'))
-------------------^
      }

  test: should follow a redirect chain up to the allowed
number of times
  stack: |
    RedirectHandler.onHeaders
(lib/handler/RedirectHandler.js:106:20)
    Request.onHeaders (lib/core/request.js:269:29)
    Parser.onHeadersComplete (lib/client.js:924:27)
    wasm_on_headers_complete (lib/client.js:536:30)
    null.<anonymous> (wasm://wasm/0003626a:1:1173)
    null.<anonymous> (wasm://wasm/0003626a:1:4100)
    null.<anonymous> (wasm://wasm/0003626a:1:28303)
    null.<anonymous> (wasm://wasm/0003626a:1:5481)
    null.<anonymous> (wasm://wasm/0003626a:1:1363)
    Parser.execute (lib/client.js:677:22)
  at:
    line: 106
    column: 20
    file: lib/handler/RedirectHandler.js
    function: RedirectHandler.onHeaders
  tapCaught: returnedPromiseRejection

test/redirect-request.js 1> History Length: 2
test/redirect-request.js 1> Max Redirects: 2
test/redirect-request.js 1> Redirection Limit Reached: true
test/redirect-request.js 1> History Length: 2
test/redirect-request.js 1> Max Redirects: 2
test/redirect-request.js 1> Redirection Limit Reached: true
```